src/Entities/Entity.cpp

```diff
@@ -1259,10 +1259,349 @@ void cEntity::DetectCacti(void)
+					}
+				}
+			}
+		}
+
+		if (Possibilities.size() > 0)
+		{
+			m_BuildPlatform = false;
+
+			// Find the nearest
+			double DistanceToClosest = (Possibilities[0] - m_Position).SqrLength();
+			Vector3d Closest = Possibilities[0];
+			for (std::vector<Vector3i>::iterator itr = Possibilities.begin(); itr != Possibilities.end(); itr++)
+			{
+				double Distance = ((*itr) - m_Position).SqrLength();
+				if (Distance < DistanceToClosest)
+				{
+					DistanceToClosest = Distance;
+					Closest = *itr;
+				}
+			}
+
+			m_PortalLoc = Closest;
+		}
+	}
+	return true;
+}
+
+
+
+void cNetherPortalScanner::BuildNetherPortal(Vector3i a_Location, int a_Direction, int a_IncludePlatform)
+{
+	int x = a_Location.x;
+	int y = a_Location.y;
+	int z = a_Location.z;
+
+	// Clear a 3x4x4 area starting right above the base
+	for (int i = 0; i < SearchSolidBaseWidth; i++)
+	{
+		for (int j = 0; j < PortalLength; j++)
+		{
+			for (int k = 1; k < PortalHeight; k++)
+			{
+				if (a_Direction)
+				{
+					m_World->SetBlock(x + i, y + k, z + j, E_BLOCK_AIR, 0);
+				}
+				else
+				{
+					m_World->SetBlock(x + j, y + k, z + i, E_BLOCK_AIR, 0);
+				}
+			}
+		}
+	}
+
+	// Put in an obsidian base
+	if (a_IncludePlatform)
+	{
+		for (int j = 0; j < PortalLength; j++)
+		{
+			if (a_Direction)
+			{
+				m_World->SetBlock(x, y, z + j, E_BLOCK_OBSIDIAN, 0);
+			}
+			else
+			{
+				m_World->SetBlock(x + j, y, z, E_BLOCK_OBSIDIAN, 0);
+			}
+		}
+	}
+
+	// Build an obsidian frame
+	for (int i = 0; i < PortalHeight; i++)
+	{
+		if (a_Direction)
+		{
+			m_World->SetBlock(x + 1, y + i, z, E_BLOCK_OBSIDIAN, 0);
+			m_World->SetBlock(x + 1, y + i, z + 3, E_BLOCK_OBSIDIAN, 0);
+		}
+		else
+		{
+			m_World->SetBlock(x, y + i, z + 1, E_BLOCK_OBSIDIAN, 0);
+			m_World->SetBlock(x + 3, y + i, z + 1, E_BLOCK_OBSIDIAN, 0);
+		}
+	}
+	for (int i = 0; i < PortalLength; i++)
+	{
+		if (a_Direction)
+		{
+			m_World->SetBlock(x + 1, y + 4, z + i, E_BLOCK_OBSIDIAN, 0);
+			m_World->SetBlock(x + 1, y, z + i, E_BLOCK_OBSIDIAN, 0);
+		}
+		else
+		{
+			m_World->SetBlock(x + i, y + 4, z + 1, E_BLOCK_OBSIDIAN, 0);
+			m_World->SetBlock(x + i, y, z + 1, E_BLOCK_OBSIDIAN, 0);
+		}
+	}
+
+	// Fill the frame (place a fire in the bottom)
+	m_World->SetBlock(x + 1, y + 1, z + 1, E_BLOCK_FIRE, 0);
+}
+
+
+
+void cNetherPortalScanner::OnDisabled(void)
+{
+	// Now we actually move the player
+	if (!m_FoundPortal)
+	{
+		// Build a new nether portal.
+		LOGD("Building nether portal at %f,%f,%f", m_PortalLoc.x, m_PortalLoc.y, m_PortalLoc.z);
+		BuildNetherPortal(m_PortalLoc, m_Dir, m_BuildPlatform);
+		m_PortalLoc.x += 1.0;
+		m_PortalLoc.y += 2.0;
+		m_PortalLoc.z += 1.0;
+	}
+
+	// Put the entity near the opening
+	if (m_Dir == 1)
+	{
+		m_PortalLoc.x += OutOffset;
+		m_PortalLoc.z += AcrossOffset;
+	}
+	else
+	{
+		m_PortalLoc.x += AcrossOffset;
+		m_PortalLoc.z += OutOffset;
+	}
+
+	m_Entity->ScheduleMoveToWorld(m_World, m_PortalLoc);
+	delete this;
+}
```

Comment:
There's no vcproj file, there's only a .vcproj.user file which is a template for per-user settings in VS. You only need to add the file to CMakeLists (it will be added to AllFiles.lst automatically)